Makefile

```diff
@@ -0,0 +1,106 @@
+# Makefile macros (or variables) are defined a little bit differently than traditional bash, keep in mind that in the Makefile there's top-level Makefile-only syntax, and everything else is bash script syntax.
+
+# .PHONY defines parts of the makefile that are not dependant on any specific file
+# This is most often used to store functions
+.PHONY: help clean build import tests
+
+# ======================
+# EXTERNAL VARIABLES
+# ======================
+
+REPOSITORY :=
+PREFIX :=
+TARGET := docker
+
+# ======================
+# INTERNAL VARIABLES
+# ======================
+
+_MAKE_DIR := .make_cache
+$(shell mkdir -p $(_MAKE_DIR))
+
+K8S_TAG := $(_MAKE_DIR)/.k8s_tag
+
+PLATFORM=amd64
+IMAGE_NAME := $(shell yq .name rockcraft.yaml)
+VERSION := $(shell yq .version rockcraft.yaml)
+
+BASE_NAME=$(IMAGE_NAME)_$(VERSION)_$(PLATFORM).tar
+
+_ROCK_OCI=$(IMAGE_NAME)_$(VERSION)_$(PLATFORM).rock
+
+_TMP_OCI_NAME := stage-$(IMAGE_NAME)
+_TMP_OCI_TAG := $(_MAKE_DIR)/$(_TMP_OCI_NAME)/$(VERSION).tag
+
+CHARMED_OCI_FULL_NAME=$(REPOSITORY)$(PREFIX)$(IMAGE_NAME)
+CHARMED_OCI_TAG := $(_MAKE_DIR)/$(CHARMED_OCI_FULL_NAME)/$(VERSION).tag
+
+help:
+	@echo "---------------HELP-----------------"
+	@echo "Image: $(IMAGE_NAME)"
+	@echo "Version: $(VERSION)"
+	@echo "Platform: $(PLATFORM)"
+	@echo " "
+	@echo "Artifact: $(BASE_NAME)"
+	@echo " "
+	@echo "Type 'make' followed by one of these keywords:"
+	@echo " "
+	@echo "  - build for creating the OCI Images"
+	@echo "  - import for importing the images to a container registry"
+	@echo "  - tests for running integration tests"
+	@echo "  - clean for removing cache file"
+	@echo "------------------------------------"
+
+$(_ROCK_OCI): rockcraft.yaml
+	@echo "=== Building Charmed Image ==="
+	rockcraft pack
+
+$(_TMP_OCI_TAG): $(_ROCK_OCI)
+	skopeo --insecure-policy \
+          copy \
+          oci-archive:"$(_ROCK_OCI)" \
+          docker-daemon:"$(_TMP_OCI_NAME):$(VERSION)"
+	if [ ! -d "$(_MAKE_DIR)/$(_TMP_OCI_NAME)" ]; then mkdir -p "$(_MAKE_DIR)/$(_TMP_OCI_NAME)"; fi
+	touch $(_TMP_OCI_TAG)
+
+$(CHARMED_OCI_TAG): $(_TMP_OCI_TAG)
+	docker build - -t "$(CHARMED_OCI_FULL_NAME):$(VERSION)" --build-arg BASE_IMAGE="$(_TMP_OCI_NAME):$(VERSION)" < Dockerfile
+	if [ ! -d "$(_MAKE_DIR)/$(CHARMED_OCI_FULL_NAME)" ]; then mkdir -p "$(_MAKE_DIR)/$(CHARMED_OCI_FULL_NAME)"; fi
+	touch $(CHARMED_OCI_TAG)
+
+$(K8S_TAG):
```

Comment:
This could do with some comments to explain what's going on. Same for `microk8s:` below

Reply:
I have added a comment in the help referring to the microk8s rule, and also placed the same comments in the different rules for better clarity